components/Typography/src/MDCFontScaler.m

```diff
@@ -0,0 +1,337 @@
+      };
+      _scalingCurve = scalingCurve;
+    } else {
+      // If nothing matches, return the metrics for MDCTextStyleBody1
+      NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+        UIContentSizeCategoryExtraSmall : @13,
+        UIContentSizeCategorySmall : @14,
+        UIContentSizeCategoryMedium : @15,
+        UIContentSizeCategoryLarge : @16,
+        UIContentSizeCategoryExtraLarge : @18,
+        UIContentSizeCategoryExtraExtraLarge : @20,
+        UIContentSizeCategoryExtraExtraExtraLarge : @22,
+        UIContentSizeCategoryAccessibilityMedium : @26,
+        UIContentSizeCategoryAccessibilityLarge : @30,
+        UIContentSizeCategoryAccessibilityExtraLarge : @34,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @38,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @42
+      };
+      _scalingCurve = scalingCurve;
+    }
+  }
+
+  return self;
+}
+
+- (UIFont *)scalableFontWithFont:(UIFont *)font {
+  // If it is available, query the preferredContentSizeCategory.
+  UIContentSizeCategory sizeCategory = UIContentSizeCategoryLarge;
+  if ([UIApplication mdc_safeSharedApplication]) {
+    sizeCategory = [UIApplication mdc_safeSharedApplication].preferredContentSizeCategory;
+  } else if (@available(iOS 10.0, *)) {
+    sizeCategory = UIScreen.mainScreen.traitCollection.preferredContentSizeCategory;
+  }
+
+  UIFont *scaledFont;
+  if (font.mdc_scalingCurve) {
+    scaledFont = [font mdc_scaledFontForSizeCategory:sizeCategory];
+    scaledFont.mdc_scalingCurve = _scalingCurve;
+  } else {
+    // The original font does not have a scaling curve, so create a copy some we aren't
+    // attaching associated objects to system fonts
+    scaledFont = [font copy];
+    scaledFont.mdc_scalingCurve = _scalingCurve;
+    scaledFont = [scaledFont mdc_scaledFontForSizeCategory:sizeCategory];
+  }
+
+  return scaledFont;
+}
+
+- (CGFloat)scaledValueForValue:(CGFloat)value {
+  // If it is available, query the preferredContentSizeCategory.
+  UIContentSizeCategory defaultSizeCategory = UIContentSizeCategoryLarge;
+  UIContentSizeCategory currentSizeCategory = UIContentSizeCategoryLarge;
+  if ([UIApplication mdc_safeSharedApplication]) {
+    currentSizeCategory = [UIApplication mdc_safeSharedApplication].preferredContentSizeCategory;
+  } else if (@available(iOS 10.0, *)) {
+    currentSizeCategory = UIScreen.mainScreen.traitCollection.preferredContentSizeCategory;
+  }
+
+  NSNumber *defaultFontSizeNumber = _scalingCurve[defaultSizeCategory];
+
+  NSNumber *currentFontSizeNumber = _scalingCurve[currentSizeCategory];
+  // If you have queried the table for a sizeCategory that doesn't exist, we will return the
+  // traits for XXXL.  This handles the case where the values are requested for one of the
+  // accessibility size categories beyond XXXL such as
+  // UIContentSizeCategoryAccessibilityExtraLarge.
+  if (currentFontSizeNumber == nil) {
```

Comment:
I'd really rather prefer that we expect all sizes to be explicit rather than this semi-magic defaulting.  

Why not also assume anything LESS than `.Large` should use the next-larger content size category that is available?  What if they provided `UIContentSizeCategoryAccessibilityMedium` but not `UIContentSizeCategoryAccessibilityLarge`? Why would it not fall-back to `.AccessibilityMedium` in that case?

Reply:
This was outdated code.
There is no longer any magic, all categories MUST be defined.
Removed.